Sources/SwiftyCrop/Models/CropViewModel.swift

```diff
@@ -34,6 +43,17 @@ class CropViewModel: ObservableObject {
         let xLimit = ((imageSizeInView.width / 2) * scale) - maskRadius
         return CGPoint(x: xLimit, y: yLimit)
     }
+    func calculateDragGestureMaxRectangle() -> CGPoint {
```

Comment:
Suggestion: Make this func private and use it in the above func `calculateDragGestureMax`. In the `CropViewModel` you should have access to the `maskShape`. Decide inside the func `calculateDragGestureMax` how to calculate based on the `maskShape`.

This also makes it easier below (I added a comment)